rotkehlchen/chain/evm/accounting/interfaces.py

```diff
@@ -92,22 +96,98 @@ def _process_deposit_or_withdrawal(
         for idx in range(events_to_consume):
             next_event = next(other_events, None)
             if next_event is None:
-                log.debug(f'Could not consume event nr. {idx} for {event.counterparty} deposit/withdrawal')  # noqa: E501
+                log.debug(
+                    f'Could not consume event nr. {idx} for {event.counterparty} deposit/withdrawal')  # noqa: E501
                 return
 
             if next_event.balance.amount == ZERO:
                 continue
 
-            pot.add_asset_change_event(
-                method=method,
-                event_type=AccountingEventType.TRANSACTION_EVENT,
-                notes=next_event.notes if next_event.notes else '',
-                location=next_event.location,
-                timestamp=next_event.get_timestamp_in_sec(),
-                asset=next_event.asset,
-                amount=next_event.balance.amount,
-                taxable=False,  # Deposits and withdrawals are not taxable
-                count_entire_amount_spend=False,
-                count_cost_basis_pnl=False,
-                extra_data={'tx_hash': next_event.serialized_event_identifier},
-            )
+            if method == 'spend':
+                amount_deposited[next_event.asset] += next_event.balance.amount
+                pot.add_asset_change_event(
+                    method=method,
+                    event_type=AccountingEventType.TRANSACTION_EVENT,
+                    notes=next_event.notes if next_event.notes else '',
+                    location=next_event.location,
+                    timestamp=next_event.get_timestamp_in_sec(),
+                    asset=next_event.asset,
+                    amount=next_event.balance.amount,
+                    taxable=False,  # Deposits and withdrawals are not taxable
+                    count_entire_amount_spend=False,
+                    count_cost_basis_pnl=False,
+                    extra_data={'tx_hash': next_event.serialized_event_identifier},
+                )
+            else:  # can only be acquisition
+                amount_deposited[next_event.asset] -= next_event.balance.amount
+                if amount_deposited[next_event.asset] < ZERO:
+                    profit = -1 * amount_deposited[next_event.asset]
```

Comment:
This doesn't cover afaik the case where you withdraw less than what you deposit. In that case you are in a lost